src/main/java/mate/academy/internetshop/dao/jdbc/ProductDaoJdbcImpl.java

```diff
@@ -0,0 +1,115 @@
+package mate.academy.internetshop.dao.jdbc;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.academy.internetshop.dao.ProductDao;
+import mate.academy.internetshop.exceptions.DataProcessingException;
+import mate.academy.internetshop.lib.Dao;
+import mate.academy.internetshop.model.Product;
+import mate.academy.internetshop.util.ConnectionUtil;
+import org.apache.log4j.Logger;
+
+@Dao
+public class ProductDaoJdbcImpl implements ProductDao {
+    private static final Logger LOGGER = Logger.getLogger(ProductDaoJdbcImpl.class);
+
+    @Override
+    public Product create(Product product) {
+        String query = "INSERT INTO products (name, price) VALUES (?, ?);";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement preparedStatement = connection
+                    .prepareStatement(query, Statement.RETURN_GENERATED_KEYS);
+            preparedStatement.setString(1, product.getName());
+            preparedStatement.setBigDecimal(2, product.getPrice());
+            if (preparedStatement.executeUpdate() > 0) {
+                ResultSet resultSet = preparedStatement.getGeneratedKeys();
+                resultSet.next();
+                product.setId(resultSet.getLong(1));
+            } else {
+                LOGGER.warn("Can't create product");
+            }
```

Comment:
it's redundant check and logging here, try to explain why in comment below

Reply:
I guess because we can't create a product only if we got SQLException then Logger never be called.

Reply:
then this exception will be logged anyway